aws/validators.go

```diff
@@ -21,6 +22,16 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
 )
 
+const (
+	awsAccountIDRegexpPattern = `^(aws|\d{12})$`
+	awsPartitionRegexpPattern = `^aws(-[a-z]+)*$`
+	awsRegionRegexpPattern    = `^[a-z]{2}(-[a-z]+)+-\d$`
```

Comment:
```suggestion
	awsRegionRegexpPattern    = `^[a-z]{2}(-[a-z]+)+-\d+$`
```
could be more future-proof

Reply:
@PatMyron indeed, however this would accept invalid values such as `us-east-01`. It would be more correct to use `-[1-9][0-9]*$` as the pattern suffix. Please feel free to submit a follow up pull request on this, if desired, since this is not a current concern with publicly documented region names.